obsplus/events/merge.py

```diff
@@ -190,6 +191,67 @@ def _associate_picks(old_eve, new_event, new_origin):
         arrival.pick_id = old_pick.resource_id
 
 
+def associate_merge(
+    event: Event,
+    new_catalog: Catalog,
+    median_tolerance: float = 1.0,
+    delete_old: bool = False,
+) -> Event:
+    """
+    Merge the "closest" event in a catalog into an existing event.
+
+    Finds the closest event in new_catalog to event using median pick
+    times, then calls :func:`obsplus.events.merge.merge_events` to merge
+    the events together.
+
+    Parameters
+    ----------
+    event
+        The base event which will be modified in place.
+    new_catalog
+        A new catalog which contains picks, or a list of Picks to
+        merge into the original event.
+    median_tolerance
+        The tolerance, in seconds, of the median pick for associating
+        events in new_catalog_or_picks into event.
+    delete_old
+        Delete any picks/amplitudes in old event if not found in new
+        event.
+    """
+
+    def _get_new_pick_df(new_catalog_or_picks):
+        if not isinstance(new_catalog_or_picks, Catalog):
+            new_cat = Catalog(events=[Event(picks=new_catalog_or_picks)])
+        else:
+            new_cat = new_catalog_or_picks
+        return obsplus.picks_to_df(new_cat)
+
+    def _get_pick_median(time_ser):
+        """Return a (close enough) approximation of the median for datetimes"""
+        int_median = int(time_ser.astype(int).median())
```

Comment:
Is the precision of this seconds or ns? If seconds, this seems kinda coarse? Though maybe that doesn't matter for most use cases.

Reply:
It is in nanoseconds, I will update the docstring to be clear on that.